WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -48,30 +55,23 @@ class LocationDetailViewController: UIViewController {
         summaryLabel.text = ""
         imageView.image = UIImage()
     }
- 
+    
     func updateUserInterface(){
         let pageViewController = UIApplication.shared.windows.first!.rootViewController as! PageViewController
```

Comment:
this code here i.e `UIApplication.shared.windows.first!.rootViewController` is generally bad code because what you are doing here is iterate through all screens in your app looking for a specific viewController. now imagine an app with 200 screens. This search is a very expensive search and you shouldn't use this.